src/reformatters/common/region_job.py

```diff
@@ -508,11 +513,11 @@ def process(self) -> Mapping[str, Sequence[SOURCE_FILE_COORD]]:
                 for data_var in data_var_group:
                     # Copy so we have a unique status per variable, not per variable group
                     data_var_source_file_coords = deepcopy(source_file_coords)
-
                     data_array, data_array_template = create_data_array_and_template(
                         processing_region_ds,
                         data_var.name,
                         shared_buffer,
+                        fill_value=data_var.encoding.fill_value,
```

Comment:
@aldenks this one slipped my mind when you asked what changes I needed to make to deal with the int16 data var.